src/pspm_sf_dcm.m

```diff
@@ -131,6 +132,26 @@
   '\n=========================================================\n'], c(4:6));
 if isfield(options, 'missing')
   ymissing = options.missing;
+  if isfile(ymissing)
+    [~, ~, fExt] = fileparts(ymissing);
+    switch lower(fExt)
+      case '.mat'
+        % A MAT file
+        ymissing = load(ymissing);
+        ymissing = ymissing.epochs;
+      otherwise  % Under all circumstances SWITCH gets an OTHERWISE!
+        warning('ID:invalid_input', 'Missing epochs should be a matlab file if loading from external.');
+    end
+  else
+    if ~isnumeric(ymissing)
+      warning('ID:invalid_input', 'Missing epochs should be numeric if as a variable.');
+    end
+  end
+  if any(size(ymissing)~=size(y))
+    ymiss = ymissing;
+    ymissing = zeros(size(y));
+    ymissing(ymiss(:,1)+1:ymiss(:,2)+1) = 1;
+  end
 else
   ymissing = isnan(y);
```

Comment:
I'm not sure this works. Above, 'ymissing' is defined by onset and offset, here it is an index?

Below it seems to be treated not as an index, but as onset and offset definition (e.g. when you use 'diff' to determine the duration of missing epochs).

Reply:
Yes, I was designing `ymissing` to be an index.
I checked the usage of `isYout` in `VBA_NLStateSpaceModel` and I find it to be an _index_. That is why I convert it from _onset and offset_ to be an _index_.
To keep the code consistent to previous versions, I can keep `ymissing` to be *onset and offset* and calculate `isYout` (which is *index*) from `ymissing`.
Sorry for making these codes confusing.